crates/l1tx/src/deposit/deposit_tx.rs

```diff
@@ -76,81 +136,239 @@ fn parse_tag_script<'a>(
         return Err(DepositParseError::TagOversized);
     }
 
-    parse_tag(data, config)
+    parse_tag(data, &config.magic_bytes, config.address_length)
 }
 
+/// Parses the script buffer which has the following structure:
+/// [magic_bytes(n bytes), stake_idx(4 bytes), ee_address(m bytes), takeback_hash(32 bytes),
+/// sats_amt(8 bytes)]
 fn parse_tag<'b>(
     buf: &'b [u8],
-    config: &DepositTxParams,
+    magic_bytes: &[u8],
+    addr_len: u8,
 ) -> Result<DepositTag<'b>, DepositParseError> {
     // data has expected magic bytes
-    let magic_bytes = &config.magic_bytes;
     let magic_len = magic_bytes.len();
 
-    // Do some math to make sure there is a magic.
-    let exp_min_len = magic_len + 4;
-    if buf.len() < exp_min_len {
-        return Err(DepositParseError::InvalidMagic);
+    if buf.len() < magic_len + DEPOSIT_IDX_LEN + SATS_AMOUNT_LEN + TAKEBACK_HASH_LEN {
```

Comment:
This check should also include the EE address length.

```suggestion
    if buf.len() < magic_len + DEPOSIT_IDX_LEN + addr_len as usize + SATS_AMOUNT_LEN + TAKEBACK_HASH_LEN {
```